packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -334,6 +366,41 @@ export class BitBucketSource extends GitBuildSource {
   }
 }
 
+/**
+ * Filter used to determine which webhooks trigger a build
+ */
+export interface WebhookFilter {
+  /**
+   * The type of webhook filter.
+   */
+  readonly type: WebhookFilterType,
+
+  /**
+   * A regular expression pattern.
+   */
+  readonly pattern: string,
+
+  /**
+   * Used to indicate that the pattern determines which webhook events do not trigger a build.
+   * If true, then a webhook event that does not match the pattern triggers a build.
+   * If false, then a webhook event that matches the pattern triggers a build.
+   *
+   * @default false
+   */
+  readonly excludeMatchedPattern?: boolean,
+}
+
+/**
+ * Filter types for webhook filters
+ */
+export enum WebhookFilterType {
+  Event = 'EVENT',
```

Comment:
What does `Event` exactly mean? Are there some limits to what can `pattern` be when `type` is `Event`? If so, I think we might be missing a modeling opportunity here...

Reply:
Different from other types, pattern for `Event` could only be one or more actions concatenated with `,`. I couldn't think of a good model to model this behavior. Please let me know if you have some idea in mind.

Refs:
[filterGroups API ref](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_CreateWebhook.html#CodeBuild-CreateWebhook-request-filterGroups)
[filterGroups user guide](https://docs.aws.amazon.com/codebuild/latest/userguide/sample-github-pull-request.html)